### PR TITLE
Add QualityIndicator and AnomalyDetector

### PR DESCRIPTION
**Pull request by balazskreith**

### CODE REVIEW

File: src/components/peerstatsview/PeerStatsView.tsx
Comment (havfo) [inline src/components/peerstatsview/PeerStatsView.tsx:1]: No need to keep the reference to the `index` in the rendering. Use the built in `index` in the map function. Also, the `index` key is not needed in the sub-elements, only the top-level one.

`stats.map((stat, index) => ...`

File: src/components/peerstatsview/QualityIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			return;
+		}
+
+		const storage = monitor.storage;
+		const windowSize = 30;
+		const stdMultiplierThreshold = 3;
+		const minDeviationThreshold = 0.05;
+		const detector = new AnomalyDetector(windowSize, stdMultiplierThreshold, minDeviationThreshold);
+
+		const listener: () => void = () => {
```
Comment (havfo) [inline src/components/peerstatsview/QualityIndicator.tsx:30]: This is never used anywhere, or am I missing something?

File: src/components/peerstatsview/PeerStatsView.tsx
Comment (balazskreith) [reply src/components/peerstatsview/PeerStatsView.tsx:1]: c7d4c43b3c385d6e65a31d4c48bd5c339676527d

File: src/components/peerstatsview/QualityIndicator.tsx (hunk 1 above)
Comment (balazskreith) [reply src/components/peerstatsview/QualityIndicator.tsx:30]: yup, you are right

**Comment by balazskreith** [comment]
Screenshot 2023-06-02 at 8 21 41

**Pull request closed without merging**